src/main/java/com/iexec/core/task/update/TaskUpdateManager.java

```diff
@@ -180,9 +163,14 @@ Task updateTaskStatusAndSave(Task task, TaskStatus newStatus) {
     Task updateTaskStatusAndSave(Task task, TaskStatus newStatus, ChainReceipt chainReceipt) {
         TaskStatus currentStatus = task.getCurrentStatus();
         task.changeStatus(newStatus, chainReceipt);
-        Task savedTask = taskRepository.save(task);
-        log.info("UpdateTaskStatus suceeded [chainTaskId:{}, currentStatus:{}, newStatus:{}]", task.getChainTaskId(), currentStatus, newStatus);
-        return savedTask;
+        Optional<Task> savedTask = taskService.updateTask(task);
+        if (savedTask.isPresent()) {
+            log.info("UpdateTaskStatus succeeded [chainTaskId:{}, currentStatus:{}, newStatus:{}]", task.getChainTaskId(), currentStatus, newStatus);
+            return savedTask.get();
+        } else {
+            log.warn("UpdateTaskStatus failed [chainTaskId:{}, currentStatus:{}, wishedStatus:{}]", task.getChainTaskId(), currentStatus, newStatus);
+            return null;
```

Comment:
Does the calling methods verify null values?

Reply:
It is never checked as return value is never used.